crates/sovereign/sovereign-chain-components/src/sovereign/types/payloads/connection.rs

```diff
@@ -16,9 +14,10 @@ pub struct SovereignConnectionOpenAckPayload {
     pub client_state: SovereignClientState,
```

Comment:
Btw, I think the client state here is supposed to be `TendermintClientState`. The payload is carrying the proof that the counterparty's client state on the chain is the same as what the counterparty expects.

We could also generalize this to `Any` to support any `Counterparty::ClientState` other than Tendermint. However, I think we can specialize it to `TendermintClientState` for now for better clarity. We will also later work on generalizing the payload builder so that one generic implementation can work for both chains.